main.tf

```diff
@@ -8,7 +8,7 @@ data "local_file" "configuration" {
 
 data "azurerm_key_vault" "certificate_vault" {
   name                = var.vault_name
-  resource_group_name = var.key_vault_resource_group
+  resource_group_name = var.env == "perftest" || var.env == "aat" ? "core-infra-${var.subscription}-rg" : "core-infra-${var.env}-rg"
```

Comment:
This won't work for Shared services as we are using ss-dev-network-rg, we pass the resource group to the module using key_vault_resource_group